neuralmonkey/trainers/generic_trainer.py

```diff
@@ -74,6 +76,8 @@ def __init__(self, objectives: List[Objective],
                              for grad, var in gradients
                              if grad is not None]
 
+            self.n_parameters = _number_of_params(gradients)
```

Comment:
Jsou to všechny trainable parametry. Ty svoje paramtery si Adam nějak počítá sám, ale neřekl bych o nich, že se trénujou. Podobně by to bylo třeba s batch normalizací - tam je zase rozptyl a střední hodnota, co se počítají, ale netrénují se. Za parametr modelu bych považoval něco, co se přímo trénuje.